front-end/src/app/shared/components/transaction-type-base/transaction-type-base.component.ts

```diff
@@ -50,45 +50,41 @@ export abstract class TransactionTypeBaseComponent implements OnInit, OnDestroy
   ) {}
 
   ngOnInit(): void {
+    this.init(this.form, this.formProperties, this.validateService, this.transactionType);
+  }
+
+  ngOnDestroy(): void {
+    this.destroy$.next(true);
+    this.destroy$.complete();
+  }
+
+  init(
+    form: FormGroup,
+    formProperties: string[],
+    validateService: ValidateService,
+    transactionType: TransactionType | undefined
+  ) {
     // Intialize FormGroup, this must be done here. Not working when initialized only above the constructor().
-    this.form = this.fb.group(this.validateService.getFormGroupFields(this.formProperties));
+    form = this.fb.group(validateService.getFormGroupFields(formProperties));
 
-    // Initialize validation tracking of current JSON schema and form data
-    this.validateService.formValidatorSchema = this.schema;
-    this.validateService.formValidatorForm = this.form;
+    // Initialize validation tracking of current JSON schema and form data.
+    validateService.formValidatorSchema = transactionType?.schema;
+    validateService.formValidatorForm = form;
 
-    // Intialize form on "Individual" entity type
-    if (this.isExisting()) {
-      const txn = { ...this.transaction } as SchATransaction;
-      this.form.patchValue({ ...txn });
-      this.form.get('entity_type')?.disable();
+    // Disable entity type form input field when editing a record.
+    if (this.isExisting(transactionType?.transaction)) {
+      const txn = { ...transactionType?.transaction } as SchATransaction;
```

Comment:
Do you want to get rid of the schema/transaction/title getters at the beginning of this component since we're not using them anymore?

Reply:
The getters/setters were getting to be a bit much when tracing the code. They all have been removed.